redis/src/cmd.rs

```diff
@@ -303,6 +303,16 @@ impl Cmd {
         self
     }
 
+    /// Used by the RedisJSON to allow the returning of an error when serialization fails on some commands
```

Comment:
From this comment it's still not clear to me exactly why we need this?

Reply:
This is the function I added because `rustc` was complaining about the `&mut Cmd` not living long enough, so instead of returning a mutable reference, it returns the command by value.

I'll update the comment to be more specific about its purpose